forgerock-authenticator/src/main/java/org/forgerock/android/auth/biometric/BiometricAuth.java

```diff
@@ -0,0 +1,217 @@
+    private final boolean allowDeviceCredentials;
+    private final FragmentActivity activity;
+    private final BiometricAuthCompletionHandler biometricAuthListener;
+
+    public static final int ERROR_NO_BIOMETRICS = -1;
+    public static final int ERROR_NO_DEVICE_CREDENTIAL = -2;
+
+    private static final String TAG = BiometricAuth.class.getSimpleName();
+
+    /**
+     * Initializes Biometric authentication with the caller and callback handlers.
+     * If device has no Biometric enrolled, it fallback to device credentials authentication.
+     * @param title the title to be displayed on the prompt.
+     * @param subtitle the subtitle to be displayed on the prompt.
+     * @param allowDeviceCredentials if {@code true}, accepts device PIN, pattern, or password to process notification.
+     * @param activity the activity of the client application that will host the prompt.
+     * @param biometricAuthListener listener for receiving the biometric authentication result.
+     */
+    public BiometricAuth(String title,
+                         String subtitle,
+                         boolean allowDeviceCredentials,
+                         @NonNull FragmentActivity activity,
+                         @NonNull BiometricAuthCompletionHandler biometricAuthListener) {
+        this.title = title;
+        this.subtitle = subtitle;
+        this.allowDeviceCredentials = allowDeviceCredentials;
+        this.activity = activity;
+        this.biometricAuthListener = biometricAuthListener;
+
+        setServicesFromActivity(activity);
+    }
+
+    /**
+     * Return the mechanism used to lock and unlock the device.
+     * @return the KeyguardManager instance.
+     */
+    public KeyguardManager getKeyguardManager() {
+        return keyguardManager;
+    }
+
+    /**
+     * The title to be displayed on the prompt.
+     * @return the Title as String.
+     */
+    public String getTitle() {
+        return title;
+    }
+
+    /**
+     * The subtitle to be displayed on the prompt.
+     * @return the Subtitle as String.
+     */
+    public String getSubtitle() {
+        return subtitle;
+    }
+
+    /**
+     * Return the Biometric Authentication listener used to return authentication result.
+     * @return the Biometric Authentication listener.
+     */
+    public BiometricAuthCompletionHandler getBiometricAuthListener() {
+        return biometricAuthListener;
+    }
+
+    /**
+     * Return the activity of the client application that will host the prompt.
+     * @return the Activity that hosts the prompt.
+     */
+    public FragmentActivity getActivity() {
+        return activity;
+    }
+
+    /*
+     * Starts authentication process.
+     */
+    public void authenticate() {
+        // API 29 and above
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            if(hasBiometricCapability()) {
+                initBiometricAuthentication();
+            }
+            return;
```

Comment:
Should we invoke the callback instead of just return? The listener method may not be invoked.